File: frontflutter/lib/features/sos/data/models/sos_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'sos_response.g.dart';

@JsonSerializable()
class SosResponse {
  final int? facilityId;
  final String? facilityName;
  final String? facilityAddress;
  final String? phone;
  final String? facilityType;
  final double? destLatitude;
  final double? destLongitude;
  final double? distanceMeters;
  final int? estimatedMinutes;
  final String status;
  final Map<String, String>? zMetadata;

  SosResponse({
    this.facilityId,
    this.facilityName,
    this.facilityAddress,
    this.phone,
    this.facilityType,
    this.destLatitude,
    this.destLongitude,
    this.distanceMeters,
    this.estimatedMinutes,
    required this.status,
    this.zMetadata,
  });

  factory SosResponse.fromJson(Map<String, dynamic> json) =>
      _$SosResponseFromJson(json);

  Map<String, dynamic> toJson() => _$SosResponseToJson(this);

  bool get isSuccess => status == 'SUCCESS';
  bool get isNoFacilityFound => status == 'NO_FACILITY_FOUND';
}

File: backjavaspring/src/main/java/com/carebrigde/backjavaspring/dto/auth/SosResponse.java
package com.carebrigde.backjavaspring.dto.auth;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data @Builder @NoArgsConstructor @AllArgsConstructor
public class SosResponse {
    private Long facilityId;
    private String facilityName;
    private String facilityAddress;
    private String phone;
    private String facilityType;
    private Double destLatitude;
    private Double destLongitude;
    private Double distanceMeters;
    private Integer estimatedMinutes;
    private String status;
    private Map<String, String> zMetadata;
}
